File: 2503-longest-subarray-with-maximum-bitwise-and/longest-subarray-with-maximum-bitwise-and.cpp
class Solution {
public:
    int longestSubarray(vector<int>& nums) {
        int ans=1;
        int n=nums.size();
        int mx=*max_element(nums.begin(),nums.end());
        int temp=1;
        for(int i=1;i<n;i++)
        {
            if(nums[i]==mx && nums[i-1]==nums[i])
            temp++;
            else
            temp=1;
            ans=max(ans,temp);
        }
        return ans;
    }
};

File: 2503-longest-subarray-with-maximum-bitwise-and/longest-subarray-with-maximum-bitwise-and.py
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        mx = max(nums, key=lambda x:x)
        ans=1
        temp=1
        n=len(nums)
        for i in range(1,n):
            if nums[i-1]==nums[i] and nums[i]==mx:
                temp+=1
            else:
                temp=1
            ans=max(ans,temp)
        return ans

        